python/lib/dependabot/python/version.rb

```diff
@@ -63,39 +65,23 @@ def <=>(other)
         local_version_comparison(other)
       end
 
-      def epoch_comparison(other)
-        unless other.is_a?(Python::Version) && other.epoch
-          return epoch.to_i.zero? ? 0 : 1
-        end
-
-        return -1 if epoch.nil?
-
-        # Epoch versions should only ever be a single number, so we can
-        # just string-comparison them.
-        return 0 if epoch.to_i == other.epoch.to_i
+      private
 
-        epoch.to_i > other.epoch.to_i ? 1 : -1
+      def epoch_comparison(other)
+        epoch.to_i - other.epoch.to_i
       end
 
       def post_version_comparison(other)
-        unless other.is_a?(Python::Version) && other.post_release_version
+        unless other.post_release_version
           return post_release_version.nil? ? 0 : 1
         end
 
         return -1 if post_release_version.nil?
 
-        # Post release versions should only ever be a single number, so we can
-        # just string-comparison them.
-        return 0 if post_release_version.to_i == other.post_release_version.to_i
-
-        post_release_version.to_i > other.post_release_version.to_i ? 1 : -1
+        post_release_version.to_i - other.post_release_version.to_i
```

Comment:
Can you elaborate on this change a little bit? It seems to work from the tests passing but I don't quite understand how it will always result in the same outcome 🙈

Reply:
Ah I was treating it like other languages' comparison functions, but I see now that Ruby strictly defines `<=>` to only return -1, 0, or 1. I'll switch this to use `<=>` which looks idiomatic.